pkg/api/v1alpha1/cloudstackmachineconfig_webhook.go

```diff
@@ -61,11 +69,38 @@ func (r *CloudStackMachineConfig) ValidateUpdate(old runtime.Object) error {
 
 	var allErrs field.ErrorList
 	allErrs = append(allErrs, validateImmutableFieldsCloudStackMachineConfig(r, oldCloudStackMachineConfig)...)
-	if len(allErrs) == 0 {
-		return nil
+
+	if len(r.Spec.DiskOffering.Id) > 0 || len(r.Spec.DiskOffering.Name) > 0 {
+		if len(r.Spec.DiskOffering.MountPath) < 2 || !strings.HasPrefix(r.Spec.DiskOffering.MountPath, "/") {
+			allErrs = append(
+				allErrs,
+				field.Invalid(field.NewPath("spec", "diskOffering", "mountPath"), r.Spec.DiskOffering.MountPath, "field is invalid"),
+			)
+		}
+		if len(r.Spec.DiskOffering.Device) == 0 {
```

Comment:
can we reuse the validation methods in api pkg?

Reply:
We can re-use the validatePath like before comments. But the current validate method implementation in api pkg is not detailed enough to tell which field is invalid. 

Let me refactor the api validate method to combine validatePath and other fields together and return detailed info so that it can re-used.

Reply:
refactored and reused.